src/services/Service.ts

```diff
@@ -0,0 +1,86 @@
+import { Headers, RequestAPI, del, get, post } from "request";
+import * as fs from "fs";
+
+interface ServiceError {
+    errorMessage: string;
+    errorCode: string;
+}
+
+export function getRequest<T>(opts: any) {
+    return new Promise<T>((resolve, reject) => {
+        get(opts, (error: any, response: any, body: any) => {
+            const parsedBody = JSON.parse(body);
+            if (error || response.statusCode < 200 || response.statusCode > 300) {
+                const errorDetails = <ServiceError>parsedBody;
+                reject(`failed with code ${response.statusCode}:\n${errorDetails.errorMessage}\n${errorDetails.errorCode}`);
+            } else {
+                resolve(parsedBody);
+            }
+        });
+    });
+}
+
+export function postRequest<T>(opts: any) {

```

Comment:
A lot of similar code in here. `Get`, `Post`, `Delete `can be handled in one function, and the `any` usage reduced with generics.